calculator.php

```diff
@@ -134,9 +129,8 @@ function finish_app($history)
     $command = choose('Are you sure to wanna quit? Yes/No ', [AGREE, DEGREE]);
 
     if ($command == AGREE) {
-        if (!$history == NULL) {
-            $history = json_encode($history);
-            file_put_contents('history.json', $history);
+        if ($history !== null) {
+            $history->filePutContent();
         }
```

Comment:
let's save new history state inside the create method of the repository class, not in the business logic code